### DISCUSSION TITLE
Third party Oauth2 token

### DISCUSSION THREAD

**Discussion by CashZarAdmin** [fief-dev/fief#365] [Q&A]
Hello everyone, 
I want to use Fief in my FastAPI project which authenticate my QuickBooks users and store their access token, refresh token and company id in my backend and provide a method to retrieve them and refresh them once I need to. Does Fief can facilitate this?

**Comment by fief-bailiff** [comment]
Hail, @CashZarAdmin 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
We indeed have a mechanism in the Admin API to retrieve a fresh access token from OAuth2 Providers. More details here: https://docs.fief.dev/going-further/oauth-provider-token/

We do not have a specific OAuth Provider for QuickBooks, but since they support [OpenID Connect](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-openid-discovery-doc), you can configure them as a Generic OpenID provider on Fief: https://docs.fief.dev/configure/oauth-providers/#generic-openid
